spec/controllers/external_authors_controller_spec.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+describe ExternalAuthorsController do
+  include LoginMacros
+  include RedirectExpectationHelper
+  let(:user) { FactoryGirl.create(:user) }
+
+  describe "GET #get_external_author_from_invitation" do
+    it "needs a valid invitaiton" do
+      get :get_external_author_from_invitation, invitation_token: "None existent"
+      it_redirects_to_with_error(root_path, "You need an invitation to do that.")
+    end
+   
+    it "needs to have a story assocated with the invitation" do
+      invitation = FactoryGirl.create(:invitation)
+      get :get_external_author_from_invitation, invitation_token: invitation.token
+      it_redirects_to_with_error(signup_path(invitation.token), "There are no stories to claim on this invitation. Did you want to sign up instead?")
+    end
+  end
+
+  describe "GET #complete_claim" do 
+    it "should give the user the works" do
+      external_author = FactoryGirl.create(:external_author)
+      user = FactoryGirl.create(:user)
+      fake_login_known_user(user)
+      invitation = FactoryGirl.create(:invitation, external_author: external_author)
+      get :complete_claim, invitation_token: invitation.token
+      it_redirects_to_with_notice(user_external_authors_path(user), "We have added the stories imported under #{external_author.email} to your account.")
+    end
+  end 
+
+  describe "GET #update" do
+    it "needs to be done by the right user" do
+      external_author = FactoryGirl.create(:external_author)
+      wrong_external_author = FactoryGirl.create(:external_author)
+      user = FactoryGirl.create(:user)
+      fake_login_known_user(user)
+      invitation = FactoryGirl.create(:invitation, external_author: wrong_external_author)
+      get :update, invitation_token: invitation.token, id: external_author.id
+      it_redirects_to_with_error(root_path, "You don't have permission to do that.")
+    end
+
+    it "can do nothing" do
```

Comment:
I _think_ this accurately describes what you want to test:
```
context "when user is right"
  context "when doing nothing with imported works"
    it "redirects with a success message"
  context "when orphaning imported works"
    it "redirects with a success message"
    context "when updating preferences"
      it "renders edit template with a success message for orphaning and an error for preferences"
```

The only thing is, line 69 doesn't specify anything that's an attribute of `external_author`. Should it maybe have `do_not_email: true` to better resemble attempting to update preferences and orphan works at the same time? (Also, I don't think it makes any practical difference, but I think it should actually be `describe "PUT #update"` and `put :update`)

You might be also able to cut back on the repetition a little by pulling out `fake_login_known_user(user)` and putting it in a `before` block.